packages/uploadthing/src/types.ts

```diff
@@ -87,26 +71,39 @@ export type GenerateUploaderOptions = {
   package: string;
 };
 
-interface UploadThingBaseResponse {
-  key: string;
-  fileName: string;
-  fileType: FileRouterInputKey;
-  fileUrl: string;
-  contentDisposition: ContentDisposition;
-  pollingJwt: string;
-  pollingUrl: string;
+/**
+ * Properties from the web File object, this is what the client sends when initiating an upload
+ */
+export interface FileUploadData {
+  name: string;
+  size: number;
+  type: string;
 }
 
-export interface PSPResponse extends UploadThingBaseResponse {
-  url: string;
-  fields: Record<string, string>;
+/**
+ * `.middleware()` can add a customId to the incoming file data
+ */
+export interface FileUploadDataWithCustomId extends FileUploadData {
+  /**
+   * As set by `.middleware()` using @link {UTFiles}
+   */
+  customId: string | null;
 }
 
-export interface MPUResponse extends UploadThingBaseResponse {
-  urls: string[];
-  uploadId: string;
-  chunkSize: number;
-  chunkCount: number;
+/**
+ * When files are uploaded, we get back a key and a URL for the file
+ */
+export interface UploadedFileData extends FileUploadDataWithCustomId {
+  key: string;
+  url: string;
 }
 
-export type UploadThingResponse = (PSPResponse | MPUResponse)[];
+/**
+ * When the client has uploaded a file and polled for data returned by `.onUploadComplete()`
+ */
+export interface ClientUploadedFileData<T> extends UploadedFileData {
+  /**
+   * Matches what's returned from the serverside `onUploadComplete` callback
+   */
+  serverData: T;
+}
```

Comment:
this is the main change i wanted to make

We start with what the client sends (FileUploadData), then once middleware has run, we may have customId (FileUploadDataWithCustomId), then when the file has uplaoded we get `key` and `url` from UT (UploadedFileData). Lastly, the client polls for server data (ClientUploadedFileData)